package/MDAnalysis/analysis/gnm.py

```diff
@@ -337,12 +353,9 @@ def generate_kirchoff(self):
             for jcounter in neighbour_atoms:
                 if jcounter > icounter and _dsq(positions[icounter], positions[jcounter]) <= cutoffsq:
                     iresidue, jresidue = residue_index_map[icounter], residue_index_map[jcounter]
-                    if self.MassWeight:
-                        contact = 1.0 / (len(self.ca.residues[iresidue].atoms) * len(self.ca.residues[jresidue].atoms)) ** 0.5
```

Comment:
also this logic appears also in the other class in this file. It would be nice if you can change it their too.